draft-yasskin-http-origin-signed-responses.md

```diff
@@ -0,0 +1,316 @@
+---
+coding: utf-8
+
+title: Origin-signed HTTP Responses
+docname: draft-yasskin-http-origin-signed-responses
+category: std
+
+ipr: trust200902
+area: gen
+workgroup: http
+keyword: Internet-Draft
+
+stand_alone: yes
+pi: [comments, sortrefs, strict, symrefs, toc]
+
+author:
+ -
+    name: Jeffrey Yasskin
+    organization: Google
+    email: jyasskin@chromium.org
+
+informative:
+  SRI: W3C.REC-SRI-20160623
+
+--- abstract
+
+This document explores how a server can send particular responses that are
+authoritative for an origin, when the server itself is not authoritative for
+that origin. For now, it focuses on the constraints covering any such mechanism.
+
+--- note_Note_to_Readers
+
+Discussion of this draft takes place on the HTTP working group mailing list
+(ietf-http-wg@w3.org), which is archived
+at <https://lists.w3.org/Archives/Public/ietf-http-wg/>.
+
+The source code and issues list for this draft can be found
+in <https://github.com/WICG/webpackage>.
+
+--- middle
+
+# Introduction
+
+When
+I
+[presented Web Packaging to DISPATCH](https://datatracker.ietf.org/doc/minutes-99-dispatch/),
+folks thought it would make sense to split it into a way to sign individual HTTP
+responses as coming from a particular origin, and separately a way to bundle a
+collection of HTTP responses. This document explores the constraints on any
+method of signing HTTP responses and briefly sketches a possible solution to the
+constraints.
+
+# Terminology
+
+Author
+: The entity that controls the server for a particular origin {{?RFC6454}}. The
+  author can get a CA to issue certificates for their private keys and can run a
+  TLS server for their origin.
+
+Intermediate
+: An entity that fetches signed HTTP responses from an author or another
+  intermediate and forwards them to another intermediate or a client.
+
+Client
+: An entity that uses a signed HTTP response and needs to be able to prove that
+  the author vouched for it as coming from its claimed origin.
+
+# Use cases
```

Comment:
@igrigorik, could you check that these performance-related use cases for signed **single** responses make sense? (https://jyasskin.github.io/webpackage/draft-yasskin-origin-signed-responses/draft-yasskin-http-origin-signed-responses.html#use-cases)

I added some explicit text to the [packaging use cases](https://wicg.github.io/webpackage/draft-yasskin-webpackage-use-cases.html) to discourage folks from thinking of them as a performance tool, but with fewer use cases overall for non-bundled signed responses, I'm left with the performance ones here.